ghost/i18n/lib/i18n.js

```diff
@@ -85,18 +88,95 @@ function generateResources(locales, ns) {
     }, {});
 }
 
+function generateThemeResources(lng, themeLocalesPath) {
+    if (!themeLocalesPath) {
+        return {
+            [lng]: {
+                theme: {}
+            }
+        };
+    }
+
+    // Get available theme locales by scanning the directory
+    let availableLocales = [];
+    try {
+        const files = fs.readdirSync(themeLocalesPath);
+        availableLocales = files
+            .filter(file => file.endsWith('.json'))
+            .map(file => file.replace('.json', ''));
+    } catch (err) {
+        // If we can't read the directory, fall back to just trying the requested locale and English
+        availableLocales = [lng, 'en'];
+    }
+
+    // Always include the requested locale and English as fallbacks
+    const locales = [...new Set([lng, ...availableLocales, 'en'])];
+
+    return locales.reduce((acc, locale) => {
+        let res;
+        // Try to load the locale file, fallback to English
+        try {
+            const localePath = path.join(themeLocalesPath, `${locale}.json`);
+            if (fs.existsSync(localePath)) {
+                // Delete from require cache to ensure fresh reads for theme files
+                delete require.cache[require.resolve(localePath)];
+                res = require(localePath);
+            } else {
+                throw new errors.IncorrectUsageError({
+                    message: `Locale file not found: ${locale}`,
+                    context: locale
+                });
+            }
+        } catch (err) {
+            // Fallback to English if it's not the locale we're already trying
+            if (locale !== 'en') {
+                try {
+                    const enPath = path.join(themeLocalesPath, 'en.json');
+                    if (fs.existsSync(enPath)) {
+                        // Delete from require cache to ensure fresh reads for theme files
+                        delete require.cache[require.resolve(enPath)];
+                        res = require(enPath);
+                    } else {
+                        res = {};
+                    }
+                } catch (enErr) {
+                    res = {};
+                }
+            } else {
+                res = {};
+            }
```

Comment:
Would it be helpful to log that the requested locale wasn't found or do you think that will be too noisy based on the number of untranslated themes?